inc/migration.class.php

```diff
@@ -538,64 +560,70 @@ public function apply_2_1_3_migration()
-                                        'id' => $device['id'],
-                                    ],
-                                );
-                                break;
-                            }
-                        }
-                    } elseif ($device['jamf_type'] === 'Computer') {
-                        // We need to query the JSS for the computer's model identifier
-                        $computer = $this->api::getComputerByID($device['jamf_items_id'], true);
-                        if ($computer !== null) {
+            foreach ($devices as $device) {
+                if ($device['jamf_type'] === 'MobileDevice') {
+                    // We can get the model identifier directly from the list of mobile devices
+                    foreach ($jss_mobiledevices as $jss_mobile) {
+                        if ((int) $jss_mobile['id'] === $device['jamf_items_id']) {
                             $this->db->update(
                                 'glpi_plugin_jamf_devices',
                                 [
-                                    'model_identifier' => $computer['hardware']['modelIdentifier'],
+                                    'model_identifier' => $jss_mobile['modelIdentifier'],
                                 ],
                                 [
                                     'id' => $device['id'],
                                 ],
                             );
+                            break;
                         }
-                    } else {
-                        $this->glpiMigration->log('Found device with invalid Jamf type (ID: ' . $device['id'] . '). Ignoring.', true);
                     }
+                } elseif ($device['jamf_type'] === 'Computer') {
+                    // We need to query the JSS for the computer's model identifier
+                    $computer = $this->api::getComputerByID($device['jamf_items_id'], true);
+                    if ($computer !== null) {
+                        $this->db->update(
+                            'glpi_plugin_jamf_devices',
+                            [
+                                'model_identifier' => $computer['hardware']['modelIdentifier'],
+                            ],
+                            [
+                                'id' => $device['id'],
+                            ],
+                        );
+                    }
+                } else {
+                    $this->glpiMigration->log('Found device with invalid Jamf type (ID: ' . $device['id'] . '). Ignoring.', true);
                 }
             }
         }
 
+
+        //create dir if needed
+        if (!is_dir(GLPI_PLUGIN_DOC_DIR . '/jamf')) {
+            try {
+                mkdir(GLPI_PLUGIN_DOC_DIR . '/jamf');
+            } catch (Exception $e) {
+                throw new Exception('Could not create directory: ' . GLPI_PLUGIN_DOC_DIR . '/jamf. ' . $e->getMessage(), $e->getCode(), $e);
+            }
```

Comment:
As `Safe\mkdir` already raises an exception, I don't think you need to use `try catch`.
```suggestion
            mkdir(GLPI_PLUGIN_DOC_DIR . '/jamf');
```

Reply:
done !